internal/app/retriever.go

```diff
@@ -0,0 +1,121 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/filecoin-project/lassie/pkg/lassie"
+	"github.com/filecoin-project/lassie/pkg/storage"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/storage/deferred"
+	trustlessutils "github.com/ipld/go-trustless-utils"
+)
+
+// Retriever is responsible for retrieving file from the network.
+type Retriever struct {
```

Comment:
new component, that has two modes of retrieval, only the cache is being used right now